File: elixir/grade-school/school.exs
defmodule School do
  def add(db, name, grade) do
    Dict.update(db, grade, [name], &(&1 ++ [name]))
  end

  def grade(db, grade) do
    Dict.get(db, grade, [])
  end

  def sort(db) do
    Dict.merge db, db, fn(_grade, roster, _roster2) ->
      Enum.sort(roster)
    end
  end
end

File: ruby/grade-school/school.rb
class School
  def initialize
    @roster = Hash.new { |h, k| h[k] = [] }
  end

  def add(name, level)
    @roster[level] << name
  end

  def grade(level)
    @roster[level].sort
  end

  def to_hash
    @roster.sort.each_with_object({}) do |(level, names), h|
      h[level] = names.sort
    end
  end
end
